api/lib/opentelemetry.rb

```diff
@@ -41,10 +42,21 @@ def handle_error(exception: nil, message: nil)
     error_handler.call(exception: exception, message: message)
   end
 
+  # Register the global tracer provider.
+  #
+  # @param [TracerProvider] provider A tracer provider to register as the
+  #   global instance.
+  def tracer_provider=(provider)
+    # TODO: should this use a Mutex?
+    # TODO: should this only allow setting the global once?
```

Comment:
I could see app owners wanting to test custom tracing within the context of a specific test file.  In that situation you might want to reconfigure the sdk with for example the in memory exporter instead of the noop exporter.  If we disallow setting the tracer provider delegate more than once the application owners will be stuck with whatever provider the application initialized with.  

That might not be enough justification to allow setting it multiple times though.  

After typing this out, I think what's probably needed is test helpers that can by-pass these interface restrictions for testing.

Reply:
If we did only allow setting it once would we be logging a warning for subsequent attempts at setting the provider delegate?

Reply:
I don't think the spec spells out the thread-safety of globals or whether or not they can be changed. Whatever we decide, it seems like the same policy should apply to all of the globals, and not just the tracer provider. In practice, I don't know that you'd actually ever set these more than once, but I'm not sure how heavy handed we want to be in enforcing that.

Reply:
I think barring a good reason to prevent users from allow setting the global more than once, we should allow for it.  As mentioned on the sig call, logging a debug level message is probably a good idea if the tracer is being replaced.